### PR TITLE
Implement soweli pi sona audio engine

### PR DESCRIPTION
**Pull request by Beyley**
Doing this on a separate branch so we can have time to test on pTyping to make sure it works

### CODE REVIEW

**Comment by Beyley** [comment]
The ManagedBass backend of soweli pi pona is fully up to the featureset of the original implementation, although the other backends arent finished, which is why only one backend is used currently

File: Furball.Engine/Engine/Timing/AudioStreamTimeSource.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+using sowelipisona;
+
+namespace Furball.Engine.Engine.Timing {
+    public class AudioStreamTimeSource : ITimeSource {
```
Comment (Eeveelution) [inline Furball.Engine/Engine/Timing/AudioStreamTimeSource.cs:4]: really tiny thing but comments would be nice, even if tbh its self explanitory

**Review by Eeveelution — CHANGES_REQUESTED** [review]
i dont take much issue with this, from what it seems it doesnt break audio compatibility with already built software much, aside from those small comments, its all good

File: Furball.Engine/Engine/Timing/AudioStreamTimeSource.cs (hunk 1 above)
Comment (Beyley) [reply Furball.Engine/Engine/Timing/AudioStreamTimeSource.cs:4]: gotta be honest i dont know how to comment things past "this is what we grab the current time from" and "here we set what to grab the time from" which is more comments on the languages syntax then how the code works

**Comment by Eeveelution** [comment]
comments in sowelipisona in general would be good to have now that i look at it, or at least the code that gets used by the end users of furball

**Comment by Beyley** [comment]
> comments in sowelipisona in general would be good to have now that i look at it, or at least the code that gets used by the end users of furball

People arent going to be reading the soweli pi sona code so see how to code with it, and all non-self explanitory code has XMLdocs
ex: a

**Comment by Beyley** [comment]
> comments in sowelipisona in general would be good to have now that i look at it, or at least the code that gets used by the end users of furball

and also explanations of what a time source is would be better suited for a wiki explaining the ins and outs of the engine

**Comment by Beyley** [comment]
In any case, i ran through and did a quick XMLdoc of all public fields in soweli pi sona

File: Furball.Engine/Engine/Timing/AudioStreamTimeSource.cs (hunk 1 above)
Comment (Eeveelution) [reply Furball.Engine/Engine/Timing/AudioStreamTimeSource.cs:4]: hmm, fair

**Review by Eeveelution — APPROVED** [review]

**Comment by Eeveelution** [comment]
sounds good,

**Pull request merged**